apps/mobile-wallet/src/features/transactionsDisplay/ConfirmedTransactionListItem.tsx

```diff
@@ -0,0 +1,89 @@
+import { findTransactionReferenceAddress } from '@alephium/shared'
+import { useUnsortedAddressesHashes } from '@alephium/shared-react'
+import dayjs from 'dayjs'
+import localizedFormat from 'dayjs/plugin/localizedFormat' // ES 2015
+import { memo } from 'react'
+import styled from 'styled-components/native'
+
+import AppText from '~/components/AppText'
+import ListItem, { ListItemProps } from '~/components/ListItem'
+import TransactionIconStyled from '~/features/transactionsDisplay/TransactionIcon'
+import TransactionListItemAmounts from '~/features/transactionsDisplay/TransactionListItemAmounts'
+import {
+  ConfirmedTransactionListItemBaseProps,
+  ConfirmedTransactionListItemSubcomponentProps
+} from '~/features/transactionsDisplay/transactionListItemTypes'
+import useTransactionIconLabel from '~/features/transactionsDisplay/useTransactionIconLabel'
+
+dayjs.extend(localizedFormat)
+
+type ConfirmedTransactionListItemProps = Partial<ListItemProps> & ConfirmedTransactionListItemBaseProps
+
+const ConfirmedTransactionListItem = memo(({ tx, refAddressHash, ...props }: ConfirmedTransactionListItemProps) => {
+  const allAddressHashes = useUnsortedAddressesHashes()
+  const referenceAddressHash = refAddressHash ?? findTransactionReferenceAddress(allAddressHashes, tx)
+
+  if (!referenceAddressHash) return null
+
+  const commonProps = { tx, refAddressHash: referenceAddressHash }
+
+  return (
+    <ListItem
+      {...props}
+      title={<TransactionLabel {...commonProps} />}
+      subtitle={<TransactionTimestamp {...commonProps} />}
+      icon={<TransactionIcon {...commonProps} />}
+      rightSideContent={<TransactionListItemAmounts {...commonProps} />}
+      expandedSubtitle
+    />
+  )
+})
+
+export default ConfirmedTransactionListItem
+
+const TransactionLabel = ({ tx, refAddressHash }: ConfirmedTransactionListItemSubcomponentProps) => {
+  const { label } = useTransactionIconLabel(tx, refAddressHash)
+
+  return label
+}
```

Comment:
Before you ask why I created such a simple component, the `refAddressHash` is required in `useTransactionIconLabel` but it might not be available in the `ConfirmedTransactionListItem` component.

Reply:
ahh.. those hooks 🙃